src/controllers/transfer/transfer.ts

```diff
@@ -141,7 +172,12 @@ export class TransferController extends EventEmitter {
 
     // Validate the amount
     if (this.selectedToken) {
-      validationFormMsgsNew.amount = validateSendTransferAmount(this.amount, this.selectedToken)
+      validationFormMsgsNew.amount = validateSendTransferAmount(
+        this.amount,
+        Number(this.maxAmount),
```

Comment:
Relying on the Number type is risky, as it uses floating-point representation, which can lead to precision issues with numbers having more than 16 decimal places.